### PR TITLE
Adapt nRF Cloud lib and asset tracker to new shadow format

### PR DESCRIPTION
**Pull request by jtguggedal**
### net: lib: nrf_cloud: Adapt to new shadow format

The shadow document for nRF Cloud devices has grown too large
to receive by the modem. Therefore we switch to listening to
a custom topic and receive only relevant parts of the shadow.
This patch implements the required changes to topic definition
and JSON parsing.

---

### applications: asset_tracker: Enable modem info

This enables the sending of modem information to the
nRF Cloud AWS shadow document.

---

### applications: asset_tracker: Make 'device' JSON key lowercase 

The nRF Cloud web interface expects the 'device' object
to be written in lowercase.

### CODE REVIEW

**Comment by jtguggedal** [comment]
@rlubos I had to add an extra commit fixing up earlier sins, so re-requested your review.

File: applications/asset_tracker/src/cloud_codec/cloud_codec.c [hunk 1]
```diff
@@ -283,7 +283,26 @@ int cloud_encode_digital_twin_data(const struct cloud_channel_data *channel,
...
+			cJSON_Delete(reported_obj);
+			/* Dummy creation failed, but we'll let it do so
+			 * silently as it's not a functionally critical error.
+			 */
+		} else {
+			ret += json_add_obj(reported_obj, "DEVICE", dummy_obj);
+		}
+	}
+
+	ret += json_add_obj(reported_obj, channel_type_str[channel->type],
```
Comment (rlubos) [inline applications/asset_tracker/src/cloud_codec/cloud_codec.c]: We could've removed the `reported_obj` just above, is it safe to use it here unconditionally?

Comment (jtguggedal) [reply applications/asset_tracker/src/cloud_codec/cloud_codec.c]: Oops, I'm dleting the wrong object... Good catch. It should be the dummy that's deleted. Actually, no need to delete it I guess, if it was never allocated and returned `NULL`

Comment (rlubos) [reply applications/asset_tracker/src/cloud_codec/cloud_codec.c]: Yes, I agree there's no need to delete it if it failed during allocation.

Comment (jtguggedal) [reply applications/asset_tracker/src/cloud_codec/cloud_codec.c]: Removed the deletion and rebased on to master.

**Comment by jtguggedal** [comment]
Not sure what's going on with the CLA thing here, CI is green except for that one.
I was going to ignore the checkpatch warning, but can fix it if anyone objects.

**Comment by rlubos** [comment]
@jtguggedal Let's ignore it, it's the CI who is faulty in this case with no doubt. I'll merge this PR.

**Pull request merged**